### PR TITLE
storage: log information about pg wal peeks and keepalives

### PR DESCRIPTION
**Pull request by guswynn**
This is add extra logging so we can better intuit what is happening when we hit this retry in production

More information here: https://github.com/MaterializeInc/materialize/issues/16484

### CODE REVIEW

**Review by petrosagg — APPROVED** [review]

**Review by benesch — CHANGES_REQUESTED** [review]
Sorry, I'd rather not rush this in! We've to date had a hard rule about not controlling system behavior with environment variables deep in the guts of the codebase; everything needs to be plumbed in via CLI flags.

**Comment by guswynn** [comment]
@benesch Ideally this would be controllable from launchdarkly, but we don't have support for that; if we need to alter these variable to investigate this incident, what alternatives do we have?

**Comment by benesch** [comment]
>  I can plumb these through with cli flags, but it seems quite heavyweight for a single incident, and increases the blast radius of the change

Right, would love to avoid testing in production whenever possible! In absolute emergencies we might have no choice but to rig something up like this, but for relatively isolated incidents that only affect one customer/source, I'd love to avoid rushing in a change like this.

LD will give us more options here, but I wonder if we want to be looking to build something even more targeted: the ability to control these options on a per-PostgreSQL source basis. E.g.:

```
CREATE SOURCE ...
FROM POSTGRES CONNECTION pg (
    MAX WAL LENGTH '2gb'
)
```

Probably don't want to expose those knobs to end users just yet, but we could rig it up so that you can only set them as the `mz_system` user.

**Comment by guswynn** [comment]
Adjusted to just add logging, as part of the investigation in https://github.com/MaterializeInc/materialize/issues/16484

**Comment by guswynn** [comment]
@benesch we may end up concluding that this needs to be a user option!

File: src/storage/src/source/postgres.rs [hunk 1]
```diff
@@ -1101,9 +1115,21 @@ impl PostgresTaskInfo {
                 },
                 PrimaryKeepAlive(keepalive) => {
                     needs_status_update = needs_status_update || keepalive.reply() == 1;
-                    if last_data_message.elapsed() > WAL_LAG_GRACE_PERIOD
-                        && keepalive.wal_end().saturating_sub(self.lsn.into()) > MAX_WAL_LAG
+
+                    // Additional logging for incident 25
+                    {
+                        // mz_storage::source::postgres=debug,info will activate this
```
Comment (petrosagg) [inline src/storage/src/source/postgres.rs:1121]: leftover comments

**Review by petrosagg — APPROVED** [review]
The logs look good!

**Review by benesch — APPROVED** [review]

**Pull request merged**